src/Kinto.elm

```diff
@@ -63,6 +64,15 @@ type Endpoint
     | RecordEndpoint BucketName CollectionName RecordId
 
 
+type alias Body =
+    Encode.Value
+
+
+type Request
+    = KintoRequest Config Endpoint Method
+    | KintoRequestWithBody Config Endpoint Method Body
```

Comment:
yeah, sometimes "making impossible states impossible" is difficult ... I'd love to find a better way though. The thing is this should only be internal to the library, it's not public facing. Well, it is, at the moment, but it could be "hidden".